### DISCUSSION TITLE
compas_wood.joints problem

### DISCUSSION THREAD

**Discussion by basaksuvakci** [ibois-epfl/compas_wood#16] [Q&A]
Hello!

I am following your lectures on EdX. In the 'Python and compas_wood, part 1' lecture you are testing the visualstudio with 
_from compas_wood.joint import joinery_
I installed the documents via Anacoda prompt but my folder doesn't read 'joint' then I installed the github file and I imported it to the Visual Studio but still it doesn't work. I have:
compas_wood.binding/datastructures/dataset etc but not joints
I attached the image of screen too. How can I solve the problem? 

![compas_wood](https://github.com/user-attachments/assets/2b0fba49-b93a-4592-ad7d-e66ccf551fc7)
Thank you for your time.
Basak

**Comment by petrasvestartas** [comment]
@basaksuvakci 

Hi,

compas_wood has been changed slightly since recordings were done. Compas1 also was rewritten to Compas2.

Did you try to run example files? I would start from there, they are quite similar.

I am also developing Rhino Plugin, I will try to record or at least to upload a tutorial in the following days.

**Reply by basaksuvakci** [reply]
Hi! 
The example files are running perfectly. So I’ll follow the steps from example files. Do you have also tutorials for nano-wood? How does it work with compas wood? When I run the test file, it didn’t work. 
Thank you for your time.

**Comment by petrasvestartas** [comment]
`How does it work with compas wood? When I run the test file, it didn’t work.`

What error do you see? And which exact file are you running?
wood_nano is C++ binding where the main source code is located. The intention was that noone would modify to have a stable compile code.

I did a small workshop for Rhino8 and compas wood. If you are also interested to run code directly in Rhino, you can follow these pages. You would only need to install compas_wood in Rhino8 through package manager:

https://petrasvestartas.github.io/compas_wood/latest/tutorials/ibois/requirements.html
https://petrasvestartas.github.io/compas_wood/latest/tutorials/ibois/tutorials.html

**Reply by basaksuvakci** [reply]
Thank you for the links. I am going to try Rhino too. 

I tried to download the code file many times with Anaconda and without Anaconda, but each time, I had some issues. I generally have two problems with Visual 
Studio. The system can't find the compas_viewer in the file. When I download with the Anaconda prompt, then Compass. Geometry is gone.

For nano_wood, when you open a folder at Visual Studio at the end there is a 'tests' file. In the 'test' section there is a code file nano_wood. So when I run it, it says it can't find the file. When I download the nano_wood as another folder then I can't integrate them or I am integrating them wrongly.
![1](https://github.com/user-attachments/assets/6d085ec7-e7ea-4e17-bb32-08985d144cfd)
![2](https://github.com/user-attachments/assets/a2d1fca7-b17d-4892-afc3-0d2137640023)
![3](https://github.com/user-attachments/assets/e3972bb0-4589-412f-a44d-1821c4edf180)

**Comment by petrasvestartas** [comment]
Did you try to follow installation instructions described here?
I would try first `pip install -e .`

https://petrasvestartas.github.io/compas_wood/latest/installation/conda.html

**Reply by basaksuvakci** [reply]
Yes I followed the instructions as you mentioned during lesson. I think I’m having some problems at installing it. Some items are not installed as it should be. I will try more to see what’s the problem.
